library/src/cache/patch_manager.rs

```diff
@@ -17,13 +17,17 @@ const PATCHES_DIR_NAME: &str = "patches";
 const PATCHES_STATE_FILE_NAME: &str = "patches_state.json";
 const PATCH_ARTIFACT_FILENAME: &str = "dlc.vmcode";
 
-#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
```

Comment:
String does not implement Copy, so we lose the ability to make this struct Copyable. This is why this PR introduces the `as_ref`s and `clone`s below.